### ISSUE TITLE
Warn on missing gen-class if namespace has -main fn

### ISSUE THREAD

**Issue by borkdude** [clj-kondo/clj-kondo#1417]
The use case for this is when generating jar files or native images. 

Other use cases that do not need `:gen-class` which may cause false positives:

1)
``` clojure
clojure -M -m foo.bar
```

As such, it's probably best to make this an optional linter.

/cc @seancorfield
